### PR TITLE
gh-107607: Update comment about utf-8 BOM being ignored

### PR DESCRIPTION
**Pull request by rscarrera27**
According to the Python codecs docs[1], Python calls UTF-8 with BOM as utf-8-sig. Therefore, using "UTF-8-Sig" seems more appropriate.

> (...) To increase the reliability with which a UTF-8 encoding can be detected, Microsoft invented a variant of UTF-8 (that Python calls "utf-8-sig")
> (...) On encoding the utf-8-sig codec will write 0xef, 0xbb, 0xbf as the first three bytes to the file.

[1] [docs.python.org/3/library/codecs.html#encodings-and-unicode](https://docs.python.org/3/library/codecs.html#encodings-and-unicode)

* Issue: gh-107607

----
:books: Documentation preview :books:: https://cpython-previews--107858.org.readthedocs.build/

### CODE REVIEW

**Comment by ambv** [comment]
Closing and re-opening to retrigger CLA checks. Sorry for the noise.

File: Doc/reference/lexical_analysis.rst [hunk 1]
```diff
@@ -98,7 +98,7 @@ which is recognized by Bram Moolenaar's VIM.
 
 If no encoding declaration is found, the default encoding is UTF-8.  In
 addition, if the first bytes of the file are the UTF-8 byte-order mark
-(``b'\xef\xbb\xbf'``), the declared file encoding is UTF-8 (this is supported,
+(``b'\xef\xbb\xbf'``), the declared file encoding is UTF-8-Sig (this is supported,
 among others, by Microsoft's :program:`notepad`).
```
Comment (terryjreedy) [inline Doc/reference/lexical_analysis.rst]:
```suggestion
among others, by Microsoft's :program:`Notepad`).
```

File: Doc/reference/lexical_analysis.rst [hunk 2]
```diff
@@ -98,8 +98,8 @@ which is recognized by Bram Moolenaar's VIM.
 
 If no encoding declaration is found, the default encoding is UTF-8.  In
 addition, if the first bytes of the file are the UTF-8 byte-order mark
-(``b'\xef\xbb\xbf'``), the declared file encoding is UTF-8 (this is supported,
-among others, by Microsoft's :program:`notepad`).
+(``b'\xef\xbb\xbf'``), the declared file encoding is UTF-8-Sig (this is supported,
```
Comment (terryjreedy) [inline Doc/reference/lexical_analysis.rst]:
```suggestion
(``b'\xef\xbb\xbf'``), the declared file encoding is UTF-8 with BOM (this is supported,
```

**Review by terryjreedy — CHANGES_REQUESTED** [review]
This is not yet ready to merge.  I am working on a comment on the issue.

**Comment by sierrasevn** [comment]
If we aren't limited to choosing just one of the terms between "UTF-8-Sig" and "UTF_8_Sig", I also agree to use "UTF-8 with BOM" which is already used in Windows, since this page is for Windows users.

**Comment by terryjreedy** [comment]
As explained in https://github.com/python/cpython/issues/107607#issuecomment-1675967835, I now think that the entire sentence in question should be replaced with "If the implicit or explicit encoding of a file is UTF-8, a UTF-8 byte-order mark (b'\xef\xbb\xbf') is ignored rather than being a syntax error." or something very close to this.

**Review by terryjreedy — APPROVED** [review]

**Pull request merged**